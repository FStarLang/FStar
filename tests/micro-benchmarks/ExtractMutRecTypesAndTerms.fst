module ExtractMutRecTypesAndTerms

let rec f (x:nat) : Type =
  if x = 0 then unit
  else y:int { y == g (x - 1) } & f (x - 1)

and g (x:nat) : nat = 
  x
