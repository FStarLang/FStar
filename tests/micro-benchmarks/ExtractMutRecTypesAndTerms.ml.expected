open Prims
type 'x f = Prims.trivial
let rec (g : Prims.nat -> Prims.nat) = fun x -> x